### PR TITLE
Tweaks to incorporate into core deployment

### PR DESCRIPTION
**Pull request by bingen**
- Rename openzeppelin remappings, so we can use different versions in core and governance (as suggested by @danielattilasimon )
- Move initial initiatives into a separate function, so we can call it in 2 txs and avoid circular dependencies when using CREATE2

### CODE REVIEW

File: src/Governance.sol [hunk 1]
```diff
@@ -112,10 +113,23 @@ contract Governance is Multicall, UserProxyFactory, ReentrancyGuard, IGovernance
         EPOCH_DURATION = _config.epochDuration;
         require(_config.epochVotingCutoff < _config.epochDuration, "Gov: epoch-voting-cutoff-gt-epoch-duration");
         EPOCH_VOTING_CUTOFF = _config.epochVotingCutoff;
+
+        if (_initiatives.length > 0) {
+            registerInitialInitiatives(_initiatives);
+        }
+    }
+
+    function registerInitialInitiatives(address[] memory _initiatives) public {
```
Comment (danielattilasimon) [inline src/Governance.sol:122]: Bit dangerous to do this without access control.

File: src/Governance.sol [hunk 2]
```diff
@@ -112,10 +113,23 @@ contract Governance is Multicall, UserProxyFactory, ReentrancyGuard, IGovernance
...
+        require(!initialInitiativesRegistered, "Initial inintiatives already registered");
+
         for (uint256 i = 0; i < _initiatives.length; i++) {
             initiativeStates[_initiatives[i]] = InitiativeState(0, 0, 0, 0, 0);
             registeredInitiatives[_initiatives[i]] = 1;
```
Comment (danielattilasimon) [inline src/Governance.sol:127]: I wonder what happens if we theoretically neglect to call this until the first few epochs pass... obviously, let's not do that, but still, have we thought about the consequences?

File: src/Governance.sol [hunk 3]
```diff
@@ -112,10 +113,23 @@ contract Governance is Multicall, UserProxyFactory, ReentrancyGuard, IGovernance
...
+
+            emit RegisterInitiative(_initiatives[i], msg.sender, 1);
```
Comment (danielattilasimon) [inline src/Governance.sol:129]: Nice, this should have always been emitted 👍

**Review by danielattilasimon — APPROVED** [review]
Looks good! I would consider adding access control to `registerInitialInitiatives()`

File: src/Governance.sol (hunk 2 above)
Comment (GalloDaSballo) [reply src/Governance.sol:127]: Just want to flag that I had edited the deploy script to allow voting on `registeredInitiatives` the way to allow that was this:
https://github.com/liquity/V2-gov/blob/549877d976781b390207e0faacbc567a614c9681/script/DeploySepolia.s.sol#L95-L96

If you alter the logic, you may have to retest with something like this test:
https://github.com/liquity/V2-gov/blob/549877d976781b390207e0faacbc567a614c9681/test/E2E.t.sol#L100

Comment (bingen) [reply src/Governance.sol:127]: I’ll fix it.

File: src/Governance.sol (hunk 3 above)
Comment (bingen) [reply src/Governance.sol:129]: @bpierre catch :wink:

File: src/Governance.sol (hunk 2 above)
Comment (danielattilasimon) [reply src/Governance.sol]: @GalloDaSballo I don't get this...

`epochStart` is `block.timestamp - VESTING_EPOCH_START`, but `VESTING_EPOCH_START` is:
https://github.com/liquity/V2-gov/blob/549877d976781b390207e0faacbc567a614c9681/script/DeploySepolia.s.sol#L46

That doesn't look right.

Comment (GalloDaSballo) [reply src/Governance.sol]: Sorry, the intended setup is this one:
https://github.com/liquity/V2-gov/blob/549877d976781b390207e0faacbc567a614c9681/test/E2E.t.sol#L91

By setting the start one epoch in the past, the initiatives are votable

That's because we enforce only one FSM invariant and have no edge case for initiatives deployed at time zero
So by setting the start one week in the past, the initially registered initiatives are votable since the start
